### PR TITLE
Fix to defect DE12563 Stress VI with large block diagram crashes on run

### PR DESCRIPTION
**Pull request by cglzaguilar**
Removed hard code limits.

### CODE REVIEW

File: test-it/testList.json [hunk 1]
```diff
@@ -31,7 +31,8 @@
         },
         "manual": {
             "tests": [
-                "StringFormatTime.via"
+                "StringFormatTime.via",
+                "TestVI_LargeBDTest.via"
```
Comment (cglzaguilar) [inline test-it/testList.json]: Running "TestVI_LargeDBTest.via" on the framework causes Padding.via to fail on the second run. Padding.via fails with "Cannot enlarge memory arrays". I did some debugging and it appears that TestVI_LargeBDTest.via is not leaking. So something else is going on: memory fragmentation? Skipping for now.

Comment (sanmut) [inline test-it/testList.json]: Could you please try to include the fix but exclude the test "TestVI_LargeBDTest.via" and see whether/how the tests fail.

Comment (cglzaguilar) [inline test-it/testList.json]: Nothing fails when the test is included, but the fix is. My current changeset is adding the TestVI_LargeBDTest.via to the manual tests.

File: source/core/TypeTemplates.cpp [hunk 2]
```diff
@@ -147,13 +147,14 @@ void TypeTemplateVisitor::VisitCluster(ClusterType* type)
     AggregateAlignmentCalculator* saveCalc = _alignmentCalculator;
     _alignmentCalculator = &calc;
 
-    TypeRef elementTypes[1000];   // TODO(PaulAustin): enforce limits or make them dynamic
     IntIndex subElementCount = type->SubElementCount();
+    TypeRef* elementTypes = new TypeRef[subElementCount];
```
Comment (sanmut) [inline source/core/TypeTemplates.cpp]: As discussed in person last week, why don't we make this a vector (as done above)?

File: source/core/TypeTemplates.cpp [hunk 3]
```diff
@@ -180,13 +181,14 @@ void TypeTemplateVisitor::VisitEquivalence(EquivalenceType* type)
...
```
Comment (sanmut) [inline source/core/TypeTemplates.cpp]: why don't we make this a vector (as done above)?

File: source/core/TypeTemplates.cpp (hunk 2 above)
Comment (cglzaguilar) [inline source/core/TypeTemplates.cpp]: Updated code to use vector instead.

File: source/core/TypeTemplates.cpp (hunk 3 above)
Comment (cglzaguilar) [inline source/core/TypeTemplates.cpp]: Updated code to use vector instead.

File: test-it/testList.json (hunk 1 above)
Comment (cglzaguilar) [inline test-it/testList.json]: I just created a defect to look into the issue later on: DE14000 Vireo: Investigate why js test framework fails when TestVI_LargeBDTest.via is included.

**Pull request merged**